File: src/main/java/com/thunderbear06/ai/task/tasks/InteractBlockTask.java
package com.thunderbear06.ai.task.tasks;

import com.thunderbear06.entity.android.AndroidEntity;
import net.minecraft.core.BlockPos;
import net.minecraft.world.InteractionHand;

public class InteractBlockTask extends MoveToBlockTask {
    private boolean complete;

    public InteractBlockTask(AndroidEntity android, double moveSpeed, BlockPos pos) {
        super(android, moveSpeed, pos);
    }

    @Override
    public String getName() {
        return "usingBlock";
    }

    @Override
    public boolean shouldTick() {
        return !complete;
    }

    @Override
    public void tick() {
        if (isInRange(2.5D)) {
            android.getLookControl().setLookAt(getTarget().getX(), getTarget().getY(), getTarget().getZ());
            android.brain.getModules().interactionModule.interactWithBlock(InteractionHand.MAIN_HAND, getTarget());
            complete = true;
        } else {
            super.tick();
        }
    }
}

File: src/main/kotlin/com/thunderbear06/ai/task/tasks/InteractBlockTask.kt
package com.thunderbear06.ai.task.tasks

import com.thunderbear06.entity.android.AndroidEntity
import net.minecraft.core.BlockPos
import net.minecraft.world.InteractionHand

class InteractBlockTask(android: AndroidEntity, moveSpeed: Double, pos: BlockPos) : MoveToBlockTask(android, moveSpeed, pos) {
    private var complete = false

    override val name: String = "usingBlock"

    override fun shouldTick(): Boolean = !complete

    override fun tick() {
        if (isInRange(2.5)) {
            android.lookControl.setLookAt(target.x.toDouble(), target.y.toDouble(), target.z.toDouble())
            android.brain.modules.interactionModule.interactWithBlock(InteractionHand.MAIN_HAND, target)
            complete = true
        } else {
            super.tick()
        }
    }
}
